### DISCUSSION TITLE
Nodepools servers custom configuration

### DISCUSSION THREAD

**Discussion by AdZoAi** [kube-hetzner/terraform-hcloud-kube-hetzner#238] [Q&A]
Hello,

Great job !! 

Hetzner answered that they are not planing to add new images! (MicroOs request).

So meanwhile would be great if we could run (aka cloud-init) a list [] of commands to have a persisten nodepools server's custom setup.

Jorge

**Comment by mysticaltech** [comment]
Hello @AdZoAi, thanks for pinging Hetzner, it's good to let them know our interest. 

What do you mean by persistent nodepools server's custom setup? As currently, we do set up the MicroOS image via cloud-init already, to make it ready to receive k3s.

**Comment by AdZoAi** [comment]
Hello 
What we would like is to have the chance to be able to pass and "append" our scripts to yours in that "cloud init" you  use, so that we do not have to run our scripts manually in each server (also less prone of errors in case of a scale up down where always all nodes in a given nodepool should have the same config).

Regards,
Jorge

**Comment by mysticaltech** [comment]
Our cloud init file is here https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/master/modules/host/templates/userdata.yaml.tpl, as it is yaml, it can be optionally generated with the `yamlencode` terraform function, which would allow better merging with your config.

When you have something working, please do not hesitate to submit a PR back if you think that's a feature others would need.

**Comment by mysticaltech** [comment]
Also please note that the node setup is done in an independent module https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/tree/master/modules/host.

**Comment by AdZoAi** [comment]
Hello,
Many thanks for the detailed info!!
